### ISSUE TITLE
Can't open NITF in tar.gz file, but can in .tar file

### ISSUE THREAD

**Issue by cstutts** [OSGeo/gdal#2753]
## Expected behavior and actual behavior.

I have a gnu zipped tar (.tar.gz) file containing a NITF file within a sub directory. I also have an uncompressed version (.tar) tar file. I can get gdal info in python on the NITF file in the .tar file, but not the .tar.gz file. The tar.gz file uses 7 zip's default settings for zipping a .tar file.

## Steps to reproduce the problem.

This works in python: gdal.Info("/vsitar//path/to/file.tar/subdir/file.NTF")
This does not work in python: gdal.Info("/vsitar//path/to/file.tar.gz/subdir/file.NTF")
Instead I get ERROR 4: '/vsitar//path/to/file.tar.gz/subdir/file.NTF' not recognized as a support file format.

## Operating system

Ubuntu 18.04.3 LTS (Windows 10 WSL version)

## GDAL version and provenance

3.0.2 manually built.
Formats includes: NITF -raster- (rw+vs): National Imagery Transmission Format

**Comment by rouault** [comment]
Works for me with
```
gdal_translate ../autotest/gcore/data/byte.tif byte.ntf
mkdir subdir
mv byte.ntf subdir
tar cvzf subdir.tar.gz subdir
gdalinfo /vsitar/subdir.tar.gz/subdir/byte.ntf
```
You'd need to provide the .tar.gz that exhibits the issue

**Comment by cstutts** [comment]
I created an example .tar and .tar.gz with a much smaller 100x100 pixel image in it to attach here, however this example worked! My original .tar is 3 GB and .tar.gz is ~3 GB. The .tar.gz continues to fail. I generated a fresh version of the .tar.gz file and it looks like gdal creates a .tar.gz.properties file. Also, I got additional information when trying to run gdal.Info on the .tar.gz file.

Warning 1: VSIFSeekL(xxx, SEEK_END) may be really slow on GZip streams.
ERROR 4: '/vsitar//path/to/file.tar.gz/subdir/file.NTF' not recognized as a support file format.

The properties file just had this info in it.
compressed_size=3067232209
uncompressed_size=3072800768

I'm wondering if gdal is having trouble decompressing the file since it's large.
There are 4 NTFs burried in different sub folders that are around  43000 x 46000 pixels each.
Do you get my error if you have a fairly large tar.gz file (~ 3 GB)

**Comment by rouault** [comment]
Ah, I strongly believe you ran into the issue of #2315 that was fixed in 3.1.0

**Comment by cstutts** [comment]
Yes, this looks like what I'm dealing with. I was able to keep adding NTFs to the tar.gz until it failed with the same error. I will try to build 3.1.0 tomorrow and see if that fixes the issue. Thanks!

**Comment by rouault** [comment]
better jump directly to 3.1.2

**Comment by rouault** [comment]
Closing assuming this is fixed by #2315

**Comment by cstutts** [comment]
Just confirming updating to 3.1.2 fixed the issue. Thanks!
